src/Commands/ContinuousIntegration/TravisCi/TravisCiCommand.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace QuickStrap\Commands\ContinuousIntegration\TravisCi;
+
+use QuickStrap\Helpers\Composer\PackageHelper;
+use QuickStrap\Helpers\PathHelper;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+class TravisCiCommand extends Command
+{
+    /** @var  string[] */
+    private static $phpVersions = array(
+        '7.0',
+        '5.6',
+        '5.5',
+        '5.4',
+        '5.3',
+        'hhvm',
+    );
+    
+    /** @var  string */
+    private $configFile = '.travis.yml';
+    
+    /** @var  TravisCiQuestionProxy */
+    private $questionProxy;
+
+    /**
+     * TravisCiCommand constructor.
+     * @param null|string $name
+     * @param TravisCiQuestionProxy|null $questionProxy
+     * @throws \Symfony\Component\Console\Exception\LogicException
+     */
+    public function __construct($name = null, TravisCiQuestionProxy $questionProxy = null)
+    {
+        parent::__construct($name);
+        
+        if (!$questionProxy) {
+            $questionProxy = new TravisCiQuestionProxy();
+        }
+        
+        $this->questionProxy = $questionProxy;
+    }
+
+    /**
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     */
+    protected function configure()
+    {
+        parent::configure();
+
+        $this->setName('ci:travis-ci');
+        $this->setDescription('create travis-ci config');
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @return int|null
+     * @throws \Symfony\Component\Console\Exception\InvalidArgumentException
+     * @throws \Symfony\Component\Console\Exception\RuntimeException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $config = array(
```

Comment:
I would pull out config building into a builder so that you can test directly changes to the config. Let the command deal with the builder as a dependency and call things like `$builder->allowFailure('jobname')`. See [ConfigurationBuilder](https://github.com/quickstrap/quickstrap/blob/master/src/Commands/TestSuites/PhpUnit/ConfigurationBuilder.php) and [ConfigurationFactory](https://github.com/quickstrap/quickstrap/blob/master/src/Commands/TestSuites/PhpUnit/ConfigurationFactory.php) for phpunit. Following a similar pattern will allow for extending the travis-ci plugin to also configure services and other commonly used options.